yt_dlp/extractor/chaturbate.py

```diff
@@ -41,10 +42,54 @@ class ChaturbateIE(InfoExtractor):
     }]
 
     _ROOM_OFFLINE = 'Room is currently offline'
+    _ROOM_PRIVATE = 'Room is currently in a private show'
+    _ROOM_AWAY = 'Performer is currently away'
+    _ROOM_PASSWORD = 'Room is password protected'
+    _ROOM_HIDDEN = 'Hidden session in progress'
+    _ROOM_BLOCKED = 'Room is not avaiable in this region'
 
-    def _real_extract(self, url):
-        video_id, tld = self._match_valid_url(url).group('id', 'tld')
+    def _extract_from_api(self, video_id, tld):
+        req = Request(
+            f'https://chaturbate.{tld}/get_edge_hls_url_ajax/',
+            data=f'room_slug={video_id}'.encode(),
+            headers={
+                'X-Requested-With': 'XMLHttpRequest',
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+        )
+        response = self._download_json(req, video_id, fatal=False)
```

Comment:
- we can inline all the request params in the `_download_json()` call
- `'Content-Type': 'application/x-www-form-urlencoded'` header is added by default if the request has `data`
- let's future-proof this by adding the `'Accept': 'application/json'` header
- don't forget our `geo_verification_headers`, or else this would be inherently inferior to webpage extraction
- import `urlencode_postdata` from utils
- add `or {}` so that `response.get('room_status')` below is not fatal
```suggestion
        response = self._download_json(
            f'https://chaturbate.{tld}/get_edge_hls_url_ajax/', video_id,
            data=urlencode_postdata({'room_slug': video_id}), headers={
                **self.geo_verification_headers(),
                'Accept': 'application/json',
                'X-Requested-With': 'XMLHttpRequest',
            }, fatal=False) or {}
```

Reply:
done as suggested